packages/cubejs-dremio-driver/driver/DremioDriver.js

```diff
@@ -80,11 +84,24 @@ class DremioDriver extends BaseDriver {
         getEnv('dbPollMaxInterval', { dataSource })
       ) * 1000,
     };
-    const protocol = (this.config.ssl === true || this.config.ssl === 'true')
-      ? 'https'
-      : 'http';
-    this.config.url =
-      `${protocol}://${this.config.host}:${this.config.port}`;
+
+    if (this.config.dbUrl) {
+      this.config.url = this.config.dbUrl;
+      this.config.apiVersion = '';
+      this.config.auth = 'PAT';
+    } else {
+      const protocol = (this.config.ssl === true || this.config.ssl === 'true')
+        ? 'https'
+        : 'http';
+      this.config.url = `${protocol}://${this.config.host}:${this.config.port}`;
+      this.config.apiVersion = '/api/v3';
+    }
+
+    if (this.config.auth === 'PAT' || this.config.password.startsWith('Bearer ')) {
```

Comment:
These if-else blocks can be united into one with all the required checks and var sets.

Reply:
I ended up removing the `auth` variable and just check for the dremioAuthToken instead.